diff --git a/node_modules/latest-torbrowser-version/index.js b/node_modules/latest-torbrowser-version/index.js
index 6745565..452d49b 100644
--- a/node_modules/latest-torbrowser-version/index.js
+++ b/node_modules/latest-torbrowser-version/index.js
@@ -78,7 +78,7 @@ function checkPlatformSupported(version, platform) {
   const url = `${base}/${version}`;
 
   if (platform === 'darwin') {
-    platform = 'osx64';
+    platform = 'macos';
   }
 
   return new Promise((resolve, reject) => {
